Sources/Starknet/Data/Messages.swift

```diff
@@ -1,11 +1,27 @@
 import Foundation
 
 public struct MessageToL1: Decodable, Equatable {
+    public let fromAddress: Felt
     public let toAddress: Felt
-    public let payload: Felt
+    public let payload: [Felt]
 
     enum CodingKeys: String, CodingKey {
+        case fromAddress = "from_address"
         case toAddress = "to_address"
         case payload
     }
 }
+
+public struct MessageFromL1: Codable, Equatable {
+    public let fromAddress: Felt
+    public let toAddress: Felt
+    public let selector: Felt
+    public let payload: [Felt]
+
```

Comment:
Please create a public constructor

Reply:
Do you have some sort of specific public constructor in mind?
It has a default public named constructor and I don't really think we should have a constructor with unnamed arguments for this struct.

Reply:
In order to use the struct from another module you have to provide a public constructor